src/wp-includes/blocks/archives.php

```diff
@@ -69,13 +69,22 @@ function render_block_core_archives( $attributes ) {
 		$label = esc_attr( $label );
 
 		$block_content = '<label class="screen-reader-text" for="' . $dropdown_id . '">' . $title . '</label>
-	<select id="' . $dropdown_id . '" name="archive-dropdown" onchange="document.location.href=this.options[this.selectedIndex].value;">
+	<select id="' . $dropdown_id . '" name="archive-dropdown">
 	<option value="">' . $label . '</option>' . $archives . '</select>';
 
+	$js = <<<JS
+document.addEventListener( 'DOMContentLoaded', function () {
+		document.getElementById( '$dropdown_id' ).addEventListener( 'change', function () {
```

Comment:
`$dropdown_id` probably needs `esc_js`?

Reply:
I reverted this change, since the file is generated by grunt and taken from Gutenberg.